### PR TITLE
Search-154 & Search-116 (Changed encryption implementation)

### PR DESCRIPTION
**Pull request by keerthi16**
### Changed encryption implementation

- Passing the key and the userId from the client app
- Creating a encrypted file as soon as the batch indexes are merged
- Deleting the data folder when the closes

@VikasShashidhar @jay-k & @serkanmulayim Please review. Thanks

### CODE REVIEW

File: js/cryptoLib/index.js [hunk 1]
```diff
@@ -142,36 +132,37 @@ class Crypto {
...
             }
         });
     }
 
     /**
+     * Deleting the data index folder
+     * when the app is closed
+     */
+    deleteFolders() {
+        Crypto.deleteFolderRecursive(this.dataFolder);
```
Comment (VikasShashidhar) [inline js/cryptoLib/index.js]: Why do we need to delete folders recursively?

My assumption is -> While closing the app, we would only have one user's directory in the data folder, correct?

File: js/search/search.js [hunk 2]
```diff
@@ -45,17 +51,27 @@ class Search {
...
-        this.init();
+        this.crypto = new Crypto(userId, key);
+        this.decryptAndInit();
+    }
+
+    decryptAndInit() {
+        this.crypto.decryption().then(() => {
+            this.init();
+        }).catch(() => {
+            this.init();
```
Comment (VikasShashidhar) [inline js/search/search.js]: What's exactly going on here?

Comment (keerthi16) [reply js/search/search.js]: Ensuring the search library is initialised even if the decryption fails

File: js/cryptoLib/index.js (hunk 1 above)
Comment (keerthi16) [reply js/cryptoLib/index.js]: We cannot delete a directory that is not empty, So to remove a dir we need to recursively remove all the files and folder inside it.

**Pull request merged**